Automattic/wp-calypso
# Jetpack Settings: Add actions, reducer, selectors and query components for Jumpstart

### Introduction

This PR introduces actions, reducer and selectors for managing Jetpack Jumpstart. Also, a query component for managing network requests for it is being introduced. This PR is part of the Jetpack Module Settings group effort (#9171).

### New WP.COM API methods
For interacting with the .com API it introduces 2 methods that are not yet implemented on the API side:

* `getJetpackJumpstart()`
* `updateJetpackJumpstart()`

In order to fully test these methods you can:
* Apply D3588-code on your .com sandbox
* Apply https://github.com/Automattic/jetpack/pull/5813 on your Jetpack site (this PR is needed to enable the endpoint for requesting the current Jumpstart status, and to allow remote requests to the Jetpack REST API).
* Connect your Jetpack site to your .com sandbox and verify they talk to each other.

### To test

* Checkout this branch
* Play with the `QueryJetpackJumpstart` query component and verify it dispatches the proper network requests - `<QueryJetpackJumpstart siteId={ 1234 } />`.
* Verify all tests are working: `npm run test-client client/state/jetpack-settings/jumpstart/`

/cc @oskosk @roccotripaldi @johnHackworth
